### PR TITLE
Enabling setting XRT install prefix to custom location on Linux build…

### PR DESCRIPTION
**Pull request by zhangchiming**
Problem solved by the commit
Enabled setting XRT install prefix to custom location on Linux build.

Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
https://github.com/Xilinx/XRT/issues/6783

How problem was solved, alternative solutions (if any) and why they were rejected
Modified corresponding CMake files, Makefiles and build scripts.

Risks (if any) associated the changes in the commit
Low

What has been tested and how, request additional testing if necessary
Built XRT on Ubuntu 22.04.

Documentation impact (if any)
NA

### CODE REVIEW

File: build/board.sh [hunk 1]
```diff
@@ -10,7 +10,9 @@ XRTBUILD=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
 # Environment
 ################################################################
 xrt=$XRTBUILD/Release/opt/xilinx/xrt
-
+if [ x"${XRT_INSTALL_PREFIX}" != "x" ]; then
+    xrt="$XRTBUILD/Release"$XRT_INSTALL_PREFIX"/xrt"
+fi
```
Comment (stsoe) [inline build/board.sh:15]: Where does XRT_INSTALL_PREFIX come from?  I don't see it is this script?  
board.sh is a very old legacy script, I don't think it needs to support anything but the default install prefix, I say we revert the changes in this one file.

File: build/build.sh [hunk 2]
```diff
@@ -232,6 +239,10 @@ edge_dir=${EDGE_DIR:-Edge}
 # Disable with '-disable-werror' option.
 cmake_flags+=" -DXRT_ENABLE_WERROR=$werror"
 
+# set CMAKE_INSTALL_PREFIX
+export XRT_INSTALL_PREFIX=$xrt_install_prefix
```
Comment (stsoe) [inline build/build.sh:243]: Why export?  This script is not sourced, it is used only for invoking the build.  Isn't it sufficient to set CMAKE_INSTALL_PREFIX to xrt_install_prefix?  Do we need the XRT_INSTALL_PREFIX variable at all?

**Review by stsoe — COMMENTED** [review]
Thanks for working on relocation, this is long overdue.
I have a fundamental question regarding env XRT_INSTALL_PREFIX.  I didn't get too far reviewing all changes, I need to understand the reason for this external ENV var first.

File: build/board.sh (hunk 1 above)
Comment (zhangchiming) [reply build/board.sh:15]: Ok. I will remove it.

File: build/build.sh (hunk 2 above)
Comment (zhangchiming) [reply build/build.sh:243]: I thought it would be simpler to have a separate variable XRT_INSTALL_PREFIX for XRT install location. I will modify it to use CMAKE_INSTALL_PREFIX only.

**Comment by abidihaider** [comment]
Hi @zhangchiming, 

I just wanted to reach out and ask about this pull request. We have been trying to get XRT in one of research framework at CERN and this hardcoded path has been a huge blocker on our end. So we were pretty excited to see that you were able to figure this out? However, I see this request being closed. Are there plans from your side to continue and still push this fix in?

**Comment by abidihaider** [comment]
> > Hi @zhangchiming,
> > I just wanted to reach out and ask about this pull request. We have been trying to get XRT in one of research framework at CERN and this hardcoded path has been a huge blocker on our end. So we were pretty excited to see that you were able to figure this out? However, I see this request being closed. Are there plans from your side to continue and still push this fix in?
> 
> I run into some problem with pull request DCO verification because my email was originally set to private in github profile. "[144274849+zhangchiming@users.noreply.github.com](mailto:144274849+zhangchiming@users.noreply.github.com)" is expected sign off email even after I set the email to public. I will submit another pull request soon.

Awesome! Thank you so so much :)
